snap-core/src/main/java/org/esa/snap/core/dataio/dimap/DimapProductWriter.java

```diff
@@ -191,17 +251,17 @@ public void writeBandRasterData(Band sourceBand,
         final int sourceBandHeight = sourceBand.getRasterHeight();
         checkSourceRegionInsideBandRegion(sourceWidth, sourceBandWidth, sourceHeight, sourceBandHeight, sourceOffsetX,
                                           sourceOffsetY);
-        final boolean fullRaster = isFullRaster(sourceWidth, sourceHeight, sourceBandWidth, sourceBandHeight);
         final ImageOutputStream outputStream = getOrCreateImageOutputStream(sourceBand);
+        final boolean fullRaster = isFullRaster(sourceWidth, sourceHeight, sourceBandWidth, sourceBandHeight);
         if (useCache && (!fullRaster)) {
-            // @todo 1 tb/tb progress monitoring 2020-07-03
             final VariableCache variableCache = writeCache.get(sourceBand);
             final boolean canWrite = variableCache.update(sourceOffsetX, sourceOffsetY, sourceWidth, sourceHeight, sourceBuffer);
             if (canWrite) {
-                synchronized (outputStream){
+                synchronized (outputStream) {
                     variableCache.writeCompletedBlocks(outputStream);
                 }
             }
+            pm.worked(1);
```

Comment:
Not valid without pm.beginTask(...).
Also it should be wrapped with a try/finally block. Place pm.done() in the finally block.

Reply:
resolved - corrected progressbar handling